src/WebJobs.Script/StorageProvider/TableServiceClientProvider.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using Azure.Core;
+using Azure.Data.Tables;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal sealed class TableServiceClientProvider : StorageClientProvider<TableServiceClient, TableClientOptions>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TableServiceClientProvider"/> class that uses the registered Azure services to create a <see cref="TableServiceClient"/>.
+        /// </summary>
+        /// <param name="componentFactory">The Azure factory responsible for creating clients. <see cref="AzureComponentFactory"/>.</param>
+        /// <param name="logForwarder">Log forwarder that forwards events to ILogger. <see cref="AzureEventSourceLogForwarder"/>.</param>
+        public TableServiceClientProvider(AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder) : base(componentFactory, logForwarder)
+        {
+        }
+
+        protected override TableServiceClient CreateClient(IConfiguration configuration, TokenCredential tokenCredential, TableClientOptions options)
+        {
+            // If connection string is present, it will be honored first; bypass creating a serviceUri
+            if (!IsConnectionStringPresent(configuration))
+            {
+                var serviceUri = configuration.Get<StorageServiceUriOptions>()?.GetTableServiceUri();
+                if (serviceUri != null)
+                {
+                    return new TableServiceClient(serviceUri, tokenCredential, options);
+                }
+            }
+
+            return base.CreateClient(configuration, tokenCredential, options);
+        }
+
+        /// <summary>
+        /// An options class that constructs a storage service URI from a different properties.
+        /// These properties are specific to WebJobs, as there may be other relevant properties used downstream
+        /// to create storage clients.
+        /// <seealso cref="Microsoft.Extensions.Azure.ClientFactory" />
+        /// A storage service URI can be built using just the account name along with default
+        /// parameters for Scheme and Endpoint Suffix.
+        /// </summary>
+        internal class StorageServiceUriOptions
```

Comment:
I think we should promote this to a shared type that both Blob and Table Service Client Providers use, adding TableServiceUri/GetTableServiceUri members to the existing Type